src/Bundle/CoverallsBundle/Collector/GitInfoCollector.php

```diff
@@ -74,9 +74,9 @@ protected function collectBranch()
 
         foreach ($branchesResult as $result) {
             if (strpos($result, '* ') === 0) {
-                $exploded = explode('* ', $result, 2);
+                preg_match('~^\* (?:\(HEAD detached at )?([\w/]+)\)?~', $result, $matches);
```

Comment:
This breaks for branches which have hyphens (`-`) and other non-word characters in them. This method incorrectly extracts `my` for the branch name when the full branch name is `my-branch` for example.

Reply:
@smolevich I'll fix this and send you PR, k?

Reply:
My branch has hyphen and this code don't break

Reply:
Here's a test which replicates the problem (feel free to use it in this PR):
```php
    /**
     * @test
     */
    public function shouldParseDetachedBranch()
    {
        $branches = [
            '  TEST-123',
            '* TEST-456',
            '  TEST-789',
        ];

        $getHeadCommitValue = [];
        $gitCommand = $this->createGitCommandStubCalledHeadCommit($branches, $getHeadCommitValue);

        $object = new GitInfoCollector($gitCommand);

        $result =$object->collect();

        $this->assertEquals('TEST-456', $result->getBranch());
    }
```
I've seen that github actions checks out the code in detached HEAD `(HEAD detached at...)` but when running locally it's not detached and it returns `TEST` instead of `TEST-456`.

Reply:
this also broke prod release :(
https://travis-ci.org/github/php-coveralls/php-coveralls/jobs/733117026#L565

Reply:
I also recovered support for hyphens
https://github.com/php-coveralls/php-coveralls/pull/298